### PR TITLE
[PM-15635] Folders not working on desktop

### PR DESCRIPTION
**Pull request by gbubemismith**
## 🎟️ Tracking
https://bitwarden.atlassian.net/browse/PM-15635

## 📔 Objective
On desktop, each user isn’t getting their own folders. Any edits to folders are happening to all users, and it’s showing the same folders on every user. Browser appears to be fine.

## 📸 Screenshots

https://github.com/user-attachments/assets/3078878a-19a5-4144-abab-bb1aa39ab812

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: libs/common/src/vault/services/folder/folder.service.ts [hunk 1]
```diff
@@ -39,15 +40,11 @@ export class FolderService implements InternalFolderServiceAbstraction {
...
-  }
-
-  async clearDecryptedFolderState(userId: UserId): Promise<void> {
-    if (userId == null) {
-      throw new Error("User ID is required.");
-    }
-
-    await this.decryptedFoldersState(userId).forceValue([]);
+    return this.encryptedFoldersState(userId).combinedState$.pipe(
+      switchMap(([userId, folderData]) => {
```
Comment (shane-melton) [inline libs/common/src/vault/services/folder/folder.service.ts:44]: 💭 While I don't think its causing an issue here, its seems like a bit of a code smell that we're overriding the same `userId` identifier for both the `folderView$()` and `switchMap()` scopes. 

Do we need to use the `combinedState$`? Could we just rely on the `encryptedFolderState(userId)` to always return data for the provided `userId`?

File: libs/common/src/vault/services/folder/folder.service.ts
Comment (shane-melton) [inline libs/common/src/vault/services/folder/folder.service.ts:1]: ⚠️ The `replace` and `delete` methods in this file do not explicitly clear the `decryptedFolderState` which will lead to out of sync data between the decrypted `folderViews$` and the encrypted `folders$` observables.

**Review by shane-melton — CHANGES_REQUESTED** [review]
Just one issue with some missing clears on the `replace` and `delete` methods

File: libs/common/src/vault/services/folder/folder.service.ts (hunk 1 above)
Comment (gbubemismith) [reply libs/common/src/vault/services/folder/folder.service.ts:44]: Yeah we don't need the combinedstates anymore

**Review by shane-melton — APPROVED** [review]

File: libs/common/src/vault/services/key-state/folder.state.ts [hunk 2]
```diff
@@ -1,57 +1,23 @@
...
     clearOn: ["logout"],
   },
 );
 
-export const FOLDER_DECRYPTED_FOLDERS = new DeriveDefinition<
-  [UserId, Record<string, FolderData>],
-  FolderView[],
+export const FOLDER_DECRYPTED_FOLDERS = new UserKeyDefinition<FolderView[]>(
+  FOLDER_MEMORY,
+  "folder",
```
[the remark this answers is not in the document]
Comment (gbubemismith) [reply libs/common/src/vault/services/key-state/folder.state.ts:18]: Yeah I think changing it would help

**Review by addisonbeck — APPROVED** [review]

**Pull request merged**